R/plot_heatmap.R

```diff
@@ -0,0 +1,143 @@
+  
+  exp <- exposures(result) #extracting the exposures matrix
+  
+  if(isTRUE(proportional)){
+    exp <- sweep(exp, 2, colSums(exp), FUN = "/") #normalizing 
+  }
+  
+  if(isTRUE(scale)){
+    exp <- scale(exp)
+  }
+  
+  
+  annot <- sample_annotations(result) #Extracting annotations from musicatk object
+  ha = NULL #initilaizing empty annotation object
+  ha1 = NULL
+  
+   #checking if annotation argument correct
+  if(!is.null(annotation)){
+    if(!annotation %in% colnames(annot)) { 
+      stop("The given annotations are not present in the data") 
+    }
+    annot <- annot[,..annotation]
+    ha<- ComplexHeatmap::HeatmapAnnotation(df = annot)
+  }
+  
+  if(!is.null(subset_signatures)){
+    if(!subset_signatures %in% rownames(exp)) { 
+      stop("The given signatures are not present in the data") 
+    }
+    exp <- as.data.frame(exp)
+    exp <- subset(exp,rownames(exp) %in% subset_signatures)
+    exp <- as.matrix(exp)
+  }
+  
+  if(!is.null(subset_tumor)){
+    annot <- sample_annotations(result)
+    samps <-  subset(annot, Tumor_Subtypes == subset_tumor)
+    samps <- as.character(samps$Samples)
+    
+    exp <- as.data.frame(exp)
+    exp <- dplyr::select(exp,samps) #Selecting columns that match tumor subtype
+    exp <- as.matrix(exp)
+    
+    annot <- annot$Tumor_Subtypes
+    annot <- annot[annot == subset_tumor]
+    ha <- ComplexHeatmap::HeatmapAnnotation(df = annot)
+  }
+  
+ #If/else conditions to check if annotation object avaialble
+  if(isTRUE(show_column_names) & isTRUE(show_row_names)){
+    
+    if(is.null(ha) ){
+      ComplexHeatmap::Heatmap(exp,name = "exposures",show_column_names = TRUE,show_row_names = TRUE,...)
+    }
+    else if(!is.null(ha)){
+      ComplexHeatmap::Heatmap(exp,name = "exposures",top_annotation = ha,show_column_names = TRUE,show_row_names = TRUE,...)
+    }
+  } 
+  
+  else if(!isTRUE(show_column_names) & !isTRUE(show_row_names)){
+    
+    if(is.null(ha) ){
+      ComplexHeatmap::Heatmap(exp,name = "exposures",show_column_names = FALSE,show_row_names = FALSE,...)
+    }
+    else if(!is.null(ha)){
+      ComplexHeatmap::Heatmap(exp,name = "exposures",top_annotation = ha,show_column_names = FALSE,show_row_names = FALSE,...)
+    }
+  }
+  else if(isTRUE(show_column_names) & !isTRUE(show_row_names)){
+      
+    if(is.null(ha) ){
+      ComplexHeatmap::Heatmap(exp,name = "exposures",show_column_names = TRUE,show_row_names = FALSE,...)
+    }
+    else if(!is.null(ha)){
+      ComplexHeatmap::Heatmap(exp,name = "exposures",top_annotation = ha,show_column_names = TRUE,show_row_names = FALSE,...)
+    }
+  }
+  else if(!isTRUE(show_column_names) & isTRUE(show_row_names)){
+      
+    if(is.null(ha) ){
+      ComplexHeatmap::Heatmap(exp,name = "exposures",show_column_names = FALSE,show_row_names = TRUE,...)
+    }
+    else if(!is.null(ha)){
+      ComplexHeatmap::Heatmap(exp,name = "exposures",top_annotation = ha,show_column_names = FALSE,show_row_names = TRUE,...)
+    }
+  }
```

Comment:
Can this code be refactored to use the show_column_names and show_row_names as inputs to Heatmap? They're already boolean variables

Reply:
Yeah, I am not so sure how to do that. Is there a way to copy whatever the user gave as argument for these parameters directly as inputs to Heatmap?

Reply:
You may be overthinking it! show_column_names is TRUE or FALSE so you can just ```show_column_names = show_column_names```

Reply:
haha, I just realized how simple it was. I have fixed it in the latest commit.